### PR TITLE
Don't turn enter keypress into clicks on map

### PR DESCRIPTION
**Pull request by perliedman**
Addresses #5499 by:

* removing the general transformation of enter `keypress` into `click`
* adding a `keypress` listener for markers with bound popups

Possible downside: popups or other `click` handlers that added without `bindPopup` will no longer fire when enter is pressed on a focused marker.

I first tried to fix this in a more general way, but it ends up more of a hack, where we can still end up with a `click` event without mouse coordinates and `latlng`, exactly like #5499 again.

### CODE REVIEW

File: src/map/Map.js [hunk 1]
```diff
@@ -1300,9 +1300,9 @@ export var Map = Evented.extend({
 	_handleDOMEvent: function (e) {
 		if (!this._loaded || DomEvent.skipped(e)) { return; }
 
-		var type = e.type === 'keypress' && e.keyCode === 13 ? 'click' : e.type;
+		var type = e.type;
 
-		if (type === 'mousedown') {
+		if (type === 'mousedown' || e.type === 'keypress') {
```
Comment (egoroof) [inline src/map/Map.js]: `e.type` -> `type` ?

File: src/layer/Popup.js [hunk 2]
```diff
@@ -515,5 +517,11 @@ Layer.include({
 
 	_movePopup: function (e) {
 		this._popup.setLatLng(e.latlng);
+	},
+
+	_popupKeypressHandler: function (e) {
```
Comment (IvanSanchez) [inline src/layer/Popup.js]: I'd personally name this `_onKeyPress` instead, but it's just a matter of taste

Comment (perliedman) [inline src/layer/Popup.js]: Yep, agreed that was a weird name to come up with!

**Comment by egoroof** [comment]
Thanks!

**Pull request merged**